### ISSUE TITLE
[java] UseArraysAsList false negative with for-each loop

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3331]
**Affects PMD Version:** 6.36.0-SNAPSHOT

**Rule:** [UseArraysAsList](https://pmd.github.io/latest/pmd_rules_java_performance.html#usearraysaslist)

**Description:**

The rule doesn't consider for-each loops, only for-loops are recognized.

**Code Sample demonstrating the issue:**

```xml
    <test-code>
        <description>[java] UseArraysAsList false negative with for-each loop</description>
        <expected-problems>1</expected-problems>
        <expected-linenumbers>8</expected-linenumbers>
        <code><![CDATA[
import java.util.ArrayList;
import java.util.List;

public class UseArraysAsListFN {
    public List<String> convert(String[] arr) {
        List<String> result = new ArrayList<>();
        for (String s : arr) {
            result.add(s);  // violation expected
        }
        return result;
    }
}
        ]]></code>
    </test-code>

```

**Expected outcome:**

PMD should report a violation at line 8, but doesn't. This is a false-negative.
